include/deal.II/base/vectorization.h

```diff
@@ -1278,57 +1279,53 @@ class VectorizedArray<float>
-          __m128 u3 = _mm_loadu_ps(in + 4 * i + offsets[3 + outer]);
-          __m128 u4 = _mm_loadu_ps(in + 4 * i + offsets[4 + outer]);
-          __m128 u5 = _mm_loadu_ps(in + 4 * i + offsets[5 + outer]);
-          __m128 u6 = _mm_loadu_ps(in + 4 * i + offsets[6 + outer]);
-          __m128 u7 = _mm_loadu_ps(in + 4 * i + offsets[7 + outer]);
-          // To avoid warnings about uninitialized variables, need to initialize
-          // one variable with zero before using it.
-          __m256 t0, t1, t2, t3 = _mm256_set1_ps(0.F);
-          t0        = _mm256_insertf128_ps(t3, u0, 0);
-          t0        = _mm256_insertf128_ps(t0, u4, 1);
-          t1        = _mm256_insertf128_ps(t3, u1, 0);
-          t1        = _mm256_insertf128_ps(t1, u5, 1);
-          t2        = _mm256_insertf128_ps(t3, u2, 0);
-          t2        = _mm256_insertf128_ps(t2, u6, 1);
-          t3        = _mm256_insertf128_ps(t3, u3, 0);
-          t3        = _mm256_insertf128_ps(t3, u7, 1);
-          __m256 v0 = _mm256_shuffle_ps(t0, t1, 0x44);
-          __m256 v1 = _mm256_shuffle_ps(t0, t1, 0xee);
-          __m256 v2 = _mm256_shuffle_ps(t2, t3, 0x44);
-          __m256 v3 = _mm256_shuffle_ps(t2, t3, 0xee);
-          *reinterpret_cast<__m256 *>(
-            reinterpret_cast<float *>(&out[4 * i + 0].data) + outer) =
-            _mm256_shuffle_ps(v0, v2, 0x88);
-          *reinterpret_cast<__m256 *>(
-            reinterpret_cast<float *>(&out[4 * i + 1].data) + outer) =
-            _mm256_shuffle_ps(v0, v2, 0xdd);
-          *reinterpret_cast<__m256 *>(
-            reinterpret_cast<float *>(&out[4 * i + 2].data) + outer) =
-            _mm256_shuffle_ps(v1, v3, 0x88);
-          *reinterpret_cast<__m256 *>(
-            reinterpret_cast<float *>(&out[4 * i + 3].data) + outer) =
-            _mm256_shuffle_ps(v1, v3, 0xdd);
-        }
-      for (unsigned int i = 4 * n_chunks; i < n_entries; ++i)
-        for (unsigned int v = 0; v < 8; ++v)
-          out[i][v + outer] = in[offsets[v + outer] + i];
+      // To avoid warnings about uninitialized variables, need to initialize
+      // one variable with some old value.
+      __m512 t0, t1, t2, t3 = out[0].data;
+
+      t0 = _mm512_insertf32x4(t0, _mm_loadu_ps(in + offsets[0] + 4 * i), 0);
+      t0 = _mm512_insertf32x4(t0, _mm_loadu_ps(in + offsets[4] + 4 * i), 1);
+      t0 = _mm512_insertf32x4(t0, _mm_loadu_ps(in + offsets[8] + 4 * i), 2);
+      t0 = _mm512_insertf32x4(t0, _mm_loadu_ps(in + offsets[12] + 4 * i), 3);
+      t1 = _mm512_insertf32x4(t3, _mm_loadu_ps(in + offsets[1] + 4 * i), 0);
+      t1 = _mm512_insertf32x4(t1, _mm_loadu_ps(in + offsets[5] + 4 * i), 1);
+      t1 = _mm512_insertf32x4(t1, _mm_loadu_ps(in + offsets[9] + 4 * i), 2);
+      t1 = _mm512_insertf32x4(t1, _mm_loadu_ps(in + offsets[13] + 4 * i), 3);
+      t2 = _mm512_insertf32x4(t3, _mm_loadu_ps(in + offsets[2] + 4 * i), 0);
+      t2 = _mm512_insertf32x4(t2, _mm_loadu_ps(in + offsets[6] + 4 * i), 1);
```

Comment:
I have to admit that I have no idea what is going on here with the permutations in the indexes. :-)
 (don't worry and don't bother explaining).